gdx/src/com/badlogic/gdx/graphics/g2d/GlyphLayout.java

```diff
@@ -276,6 +275,13 @@ else if (targetWidth <= fontData.spaceXadvance * 3) //
 		height = fontData.capHeight + Math.abs(y);
 
 		// Calculate run widths and the entire layout width.
+		calculateAndSetWidths(fontData);
+
+		// Align runs to center or right of targetWidth.
+		alignRuns(targetWidth, halign);
```

Comment:
Having `calculateAndSetWidths` and `alignRuns` as private methods reduces the length of `setText` quite a bit.
Also, I was never sure if this code also changes any vaiables needed for the loop above, even if the blocks were outside the loop. It was just very hard to keep track of this monster function.